x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,327 @@
+              defaultMessage: 'field statistics',
+            }),
+          isEditingEnabled: () => true,
+          onEdit: async () => {
+            try {
+              const { resolveEmbeddableFieldStatsUserInput } = await import(
+                './resolve_field_stats_embeddable_input'
+              );
+              const chartState = serializeFieldStatsChartState();
+              const nextUpdate = await resolveEmbeddableFieldStatsUserInput(
+                coreStart,
+                pluginStart,
+                parentApi,
+                uuid,
+                chartState,
+                fieldStatsControlsApi
+              );
+
+              fieldStatsControlsApi.updateUserInput(nextUpdate);
+            } catch (e) {
+              return Promise.reject(e);
+            }
+          },
+          dataViews: dataViews$,
+          serializeState: () => {
+            const dataViewId = fieldStatsControlsApi.dataViewId$?.getValue();
+            const references: Reference[] = dataViewId
+              ? [
+                  {
+                    type: DATA_VIEW_SAVED_OBJECT_TYPE,
+                    name: FIELD_STATS_DATA_VIEW_REF_NAME,
+                    id: dataViewId,
+                  },
+                ]
+              : [];
+            return {
+              rawState: {
+                timeRange: undefined,
+                ...serializeTitles(),
+                ...serializeTimeRange(),
+                ...serializeFieldStatsChartState(),
+              },
+              references,
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          ...fieldStatsControlsComparators,
+        }
+      );
+
+      const reload$ = fetch$(api).pipe(
+        skipWhile((fetchContext) => !fetchContext.isReload),
+        map(() => Date.now())
+      );
+      const query$ = fetch$(api).pipe(
+        map((fetchContext) => fetchContext.query),
+        distinctUntilChanged(fastIsEqual)
+      );
+      const filters$ = fetch$(api).pipe(
+        map((fetchContext) => fetchContext.filters),
+        distinctUntilChanged(fastIsEqual)
+      );
+      const reset$ = resetData$.pipe(skip(1), distinctUntilChanged());
+
+      const appliedTimeRange$: Observable<TimeRange | undefined> = fetch$(api).pipe(
+        map((fetchContext) => fetchContext.timeRange),
+        distinctUntilChanged(fastIsEqual)
+      );
+
+      const onTableUpdate = (changes: Partial<DataVisualizerTableState>) => {
+        if (isDefined(changes?.showDistributions)) {
+          fieldStatsControlsApi.showDistributions$.next(changes.showDistributions);
+        }
+      };
+      const statsTableCss = css({
+        width: '100%',
+        height: '100%',
+        overflowY: 'auto',
+      });
+
+      return {
+        api,
+        Component: () => {
+          if (!apiHasExecutionContext(parentApi)) {
+            throw new Error('Parent API does not have execution context');
```

Comment:
Should not throw in component render. Instead, set blocking error when parentApi is not of the expected shape.

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/f990d2278cc309c21b0893cd076b8e2f2ec127b1